posts/2024-05-06-jakarta-nosql-in-action-meet-jnopo-game.adoc

```diff
@@ -0,0 +1,799 @@
+[INFO] [AUDIT   ] CWWKT0016I: Web application available (default_host): http://localhost:9080/jakarta-nosql-game/
+----
+
+In this blog post, let's focus on two of these url:
+
+* http://localhost:9080/jakarta-nosql-game/ : it's the url of the application that we can use locally;
+
+* http://localhost:9080/openapi/ui/ : it's the Swagger UI interface that  https://openliberty.io/[Open Liberty]  offers out-of-the-box, facilitating the rest endpoint testing via browser.
+
+[.img_border_light]
+image::/img/blog/swagger-ui-01.png[Swagger UI 01, align="center" width=80%]
+
+[.img_border_light]
+image::/img/blog/swagger-ui-02.png[Swagger UI 02, align="center" width=80%]
+
+
+If you're a command-line practitioner, you can check the data by using the `curl` command:
+
+[source, bash]
+----
+curl -X 'GET' \
+  'http://localhost:9080/jakarta-nosql-game/api/playoffs/ranking' \
+  -H 'accept: application/json' | jq
+----
+
+[source, json]
+----
+{
+  "data": {
+    "Max": 3,
+    "Ghost": 1
+  }
+}
+----
+
+Congratulations if you made it this far!
+
+Now, *JNopo* is providing a winner ranking!
+
+If you're interesting to take a look on the finished project code version developed during this challenge, clone this https://github.com/OpenLiberty/sample-jakartaNoSQL-game[Git repository] and then navigate to the `winner-ranking-challenge` directory.
+
+[source, bash]
+----
+git clone https://github.com/OpenLiberty/sample-jakartaNoSQL-game.git
+cd sample-jakartaNoSQL-game
+cd winner-ranking-challenge
+----
+
+== Key Takeaways
+
+This challenge was an amazing opportunity to learn not just about Jakarta NoSQL and Jakarta Data to simplify the NoSQL integration with Java applications but be exploring new development approaches using interesting Java features.
+
+What we got by doing these challenges:
+
+* A practical example that uses:
+** Sealed classes (final feature since Java 17)
+** Pattern Matching for Instanceof (final feature since Java 16)
+** Pattern Matching for Switch (final feature since Java 21);
+
+What we learned from them:
+
+* How to configure Eclipse JNoSQL to allow us to work with Jakarta NoSQL and Jakarta Data in a Jakarta EE / MicroProfile project
+* How to work with Jakarta NoSQL to create entities pretty similar to Jakarta Persistence approach
+* How to create repositories with Jakarta Data, the newest Jakarta EE Specification that will be available in the Jakarta EE 11
+
+== Next Steps: Continuing the Journey
+
+Congratulations on getting this far!
+
+In the next blog post, a new challenge is introduced: *Switching NoSQL Databases with Ease*! Stay tuned!
+
+This blog post is the 2nd blog in a 3-part blog series. You can check out the other blogs in this series through the links below:
+
+* link:../../../2024/04/29/simplifying-nosql-database-integration-with-jakarta-nosql.html[Simplifying NoSQL Database Integration with Jakarta NoSQL, window="_blank"];
```

Comment:
will have to manage these links to reflect new filenamaes and also not expose link to final post before it's published

Reply:
Okay! I'll keep it as is by now. Let's fix it before publishing it.